specs/flyout.md

```diff
@@ -0,0 +1,152 @@
+
+### Use Cases
+- As a Dropdown for a menu, navigation bar, etc.
+- In a context menu as the secondary menu/action.
+- As a popup showing additional info for an item on the page.
+- To display a larger image of a thumbnail image in a photo gallery.
+- To display an action related to a page item
+
+### Features
+**Placement Logic:** - This component will use *Anchored Region* to be positioned relative to the Flyouts anchor element. This way it can be dynamic and adapt to the available space if `responsive` is true for the Flyout.
+
+**Light Dismiss (aka Soft Dismiss):** - A way to quickly close the Flyout by clicking outside it, or pressing the `esc` key.
+
+**Overlay:** - A control of whether or not the Flyout has an overlay effect that dims the surrounding UI when the Flyout is active.
+
+### Prior Art/Examples
+##### Context Menu
+*Note:* the "Manage, Apply, Show more options" is the Flyout for this example.
+![Context Menu](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-nested.png)
+
+##### Text Only
+*Note:* the button is not part of the Flyout, but shown for context as an anchor element.
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+##### Actionable
+*Note:* the button is not part of the Flyout, but shown for context as an anchor element.
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+---
+
+### API
+*Component Name*
+- `fast-flyout`
+
+*Attributes*
+- `anchor` - string, the html id of the HTMLElement that the Flyout is attached to
+- `overlay` - boolean, specifies whether an overlay is drawn to dim the out of scope UI until the Flyout is dismissed
+- `light-dismiss` - boolean, specifies whether or not the flyout can be dismissed by clicking elsewhere or pressing `esc` key, defaults to true
+- `visible` - boolean, whether or not the Flyout is visible, defaults to undefined, as this is controlled by the author
+- `responsive` - boolean, whether or not the positioning is responsive based on available space, defaults to true, uses Anchored Region logic
+- `position` - enum, where the tooltip should appear relative to its target, uses Anchored Region logic
+    - top-centered
+    - right-centered
+    - bottom-centered
+    - left-centered
+    - top-left-corner
+    - top-right-corner
+    - bottom-right-corner
+    - bottom-left-corner
+    - top-left-aligned
+    - right-top-aligned
+    - bottom-left-aligned
+    - left-top-aligned
+        - **ASK (1)**
+
+*Properties*
+- `anchorElement` - the reference to the HTMLElement used as an anchor
+- `visible` - boolean, whether or not the Flyout is visible
+*Methods*
+- `isNonFocusableContent()` - check if the slotted content is a focusable element or not, could use a filter to check the node type.
```

Comment:
What's the use case for this?  Flyout could have complex content.

Reply:
Yes, but if Flyout does NOT have complex/focusable content we need to ensure that the flyout is still accessible by making the flyout itself focusable.

Reply:
Is there an easy/reliable way to detect that with complex flyout content?  What if it changes during the lifetime of the flyout?  Wondering if this is a decision we could let the author make.

Reply:
This should be the same behavior as Dialog I believe; do we need a method for this? If we need to set and trap focus for modal instances we'll be querying already for this. If there are no focusable elements I'd assume we'd set focus here. Not sure we need a method for authors to call here.

Reply:
Could we not let authors decide this somehow rather than trying to detect it?

Reply:
This should likely have a role of "dialog" so I assume we'll want to handle this in the same way - I'm not sure why we would treat it differently.